### PR TITLE
[12.0][ADD] purchase_security

### PR DESCRIPTION
**Pull request by joao-p-marques**
Create "Purchase (own orders)" group.
Restrict access to POs to those in the group.
Add tests.

TT26488

### CODE REVIEW

**Comment by joao-p-marques** [comment]
ping @rafaelbn

**Review by rafaelbn — APPROVED** [review]
Tested functionally in runbot :+1:

File: purchase_security/README.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/142/13.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This addon creates a new group called "Purchase (own orders)" in Purchase.
+
+For users in this group, the only Purchase Orders they can see are those where
+they are responsible, or all of them if they are managers.
```
Comment (pedrobaeza) [inline purchase_security/README.rst]:
```suggestion
they are the representative, or all of them if they are managers.
```

File: purchase_security/README.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+To use this module you need to add the users whose access you want to restrict
```
Comment (pedrobaeza) [inline purchase_security/README.rst]: This is more configuration.

Add another usage section where you specify where to go for setting the purchase representative.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Any reason for now allowing to see purchase orders without purchase representative the same as Odoo does with sales?

**Comment by joao-p-marques** [comment]
Changes made @pedrobaeza. Thank you!
Tests also modified accordingly. 
I didn't know that was the default Odoo behaviour. Makes sense to follow it.

**Review by pedrobaeza — APPROVED** [review]

**Comment by joao-p-marques** [comment]
@pedrobaeza seems like tests are failing for some reason not to do with this module. https://travis-ci.com/github/OCA/purchase-workflow/jobs/428610268

**Pull request merged**